brave/src/test/java/brave/secondary_sampling/integration/SecondarySamplingIntegratedTest.java

```diff
@@ -45,45 +48,56 @@
   InMemoryStorage gatewayplay = InMemoryStorage.newBuilder().build();
   InMemoryStorage authcache = InMemoryStorage.newBuilder().build();
 
+  Reporter<Span> zipkinReporter = // used only in base case tests
+    s -> zipkin.spanConsumer().accept(asList(s)).enqueue(NOOP_CALLBACK);
+
   Reporter<zipkin2.Span> traceForwarder = new TraceForwarder()
     .configureSamplingKey("b3", zipkin.spanConsumer())
     .configureSamplingKey("gatewayplay", gatewayplay.spanConsumer())
     .configureSamplingKey("authcache", authcache.spanConsumer());
 
   Propagation.Factory b3 = B3SinglePropagation.FACTORY;
 
-  TestSecondarySampler gatewayplaySampler = new TestSecondarySampler()
-    .addTrigger("gatewayplay", "gateway", new Trigger().rps("/play", 50))
-    .addTrigger("gatewayplay", "playback", new Trigger().mode(PASSIVE));
+  SamplerController gatewayplaySampler = new SamplerController.Default()
+    .putSecondaryHttpRule("gateway", "gatewayplay",
```

Comment:
@adriancole you already know this but emphasizing again that all these rules should be re-loadable dynamically, without requiring a restart or redeploy of the app. The API we expose should allow implementations to reconfigure these rules at runtime.

Reply:
this is all the test directory. I don't currently have any plans to implement a control plane or an api for it. this is just to prove the concept.

Reply:
https://github.com/openzipkin-contrib/zipkin-secondary-sampling/commit/a627d4035b36ad85b79daebc90be53d7a04fe29a

Reply:
thanks. it is important to remember roles and responsibilities. Definitely, I don't have time to make a real sampling control plane, it would take months to get the design right, and it is better to be honest that we don't know what sort of api will be generally needed. As this only controls configuration, it doesn't need to be widely defined (in other words can be undefined).

You can feel free to try and implement one, just personally I cannot. When implementing this remember control planes are really tricky. For example coordinated rates require coordination https://github.com/openzipkin-attic/zipkin-zookeeper

Reply:
fwiw dynamic configuration has been implemented for a while. The main thing is that the SecondarySampling instance itself must be fixed, as the static configuration pass reads it once. This test shows that you don't need to "reboot" your graph to change policy https://github.com/openzipkin-contrib/zipkin-secondary-sampling/blob/master/brave/src/test/java/brave/secondary_sampling/SecondarySamplingStateTest.java#L88